Sources/Core/View/SwiftUI/SnackbarView.swift

```diff
@@ -0,0 +1,248 @@
+            self.button = {
+               button(
+                   ButtonView(
+                       theme: viewModel.theme,
+                       intent: viewModel.buttonIntent,
+                       variant: viewModel.buttonVariant,
+                       size: SnackbarConstants.buttonSize,
+                       shape: SnackbarConstants.buttonShape,
+                       alignment: .leadingImage,
+                       action: action
+                   )
+               )
+           }
+        } else {
+            self.button = { nil }
+        }
+    }
+
+    public init(
+        theme: any Theme,
+        intent: SnackbarIntent,
+        image: Image? = nil,
+        text: @escaping () -> Text
+    ) where SnackbarButton == EmptyView {
+        self.init(
+            theme: theme,
+            intent: intent,
+            type: .horizontal,
+            variant: .filled,
+            image: image,
+            text: text
+        )
+    }
+
+    public init(
+        theme: any Theme,
+        intent: SnackbarIntent,
+        image: Image? = nil,
+        text: @escaping () -> Text,
+        button: @escaping (SnackbarView.ButtonConfiguration) -> SnackbarButton
+    ) {
+        self.init(
+            theme: theme,
+            intent: intent,
+            type: .horizontal,
+            variant: .filled,
+            image: image,
+            text: text,
+            buttonConfiguration: button
+        )
+    }
+
+    public init(
+        theme: any Theme,
+        intent: SnackbarIntent,
+        image: Image? = nil,
+        text: @escaping () -> Text,
+        button: @escaping (ButtonView) -> SnackbarButton,
+        action: @escaping () -> Void
+    ) {
+        self.init(
+            theme: theme,
+            intent: intent,
+            type: .horizontal,
+            variant: .filled,
+            image: image,
+            text: text,
+            button: button,
+            action: action
+        )
+    }
+
+    public var body: some View {
+        ZStack {
+            self.viewModel.backgroundColor.color
+            VStack {
+                self.content()
+            }
+            .padding(.leading, self.viewModel.theme.layout.spacing.large)
+            .padding(.trailing, self.viewModel.theme.layout.spacing.medium)
+        }
+        .clipShape(.rect(cornerRadius: self.cornerRadius))
+        .shadow(self.viewModel.theme.elevation.dropShadow)
+        .accessibilityElement(children: .contain)
+        .accessibilityIdentifier(SnackbarAccessibilityIdentifier.view)
+    }
+
+    @ViewBuilder
+    private func content() -> some View {
+        switch self.type {
+        case .horizontal:
+            self.horizontalContent()
+        case .vertical:
+            self.verticalContent()
+        }
+    }
+
+    @ViewBuilder
+    private func horizontalContent() -> some View {
+        HStack(alignment: .center, spacing: self.viewModel.theme.layout.spacing.medium) {
```

Comment:
Same for all spacing in this file

Reply:
Can't be done because UIKit and SwiftUI have different behaviours